app/ts/components/subcomponents/SomeTimeAgo.tsx

```diff
@@ -1,31 +1,27 @@
-import { useEffect, useState } from 'preact/hooks'
+import { useEffect } from 'preact/hooks'
+import { useComputed, useSignal } from '@preact/signals'
 
 export interface SomeTimeAgoProps {
 	priorTimestamp: Date,
 	countBackwards?: boolean,
 }
 
 export function SomeTimeAgo(props: SomeTimeAgoProps) {
-	const [, setNewSecondTrigger] = useState(0)
-	useEffect(() => {
-		const id = setInterval(() => setNewSecondTrigger((old) => props.countBackwards ? old - 1 : old + 1), 1000)
-		return () => clearInterval(id)
-	}, [])
-	return <>{ props.countBackwards ?
-		humanReadableDateDelta(props.priorTimestamp.getTime(), new Date().getTime()) :
-		humanReadableDateDelta(new Date().getTime(), props.priorTimestamp.getTime())
-	}</>
+	const getTimeDiff = () => (props.priorTimestamp.getTime() - new Date().getTime()) / 1000
```

Comment:
The second useEffect could be skipped by updating the signal on every interval

```suggestion
	const timeNow = useSignal(new Date())
	const timeDiff = useComputed(() => props.priorTimestamp.getTime() - timeNow.value.getTime())
	const diffInSeconds = (timeDiff.value / 1000) * (props.countBackwards ? 1 : -1)
	const humanReadableTimeDiff = humanReadableDateDelta(diffInSeconds)

	useEffect(() => {
		const id = setInterval(() => (timeNow.value = new Date()), 1000)
		return () => clearInterval(id)
	}, [])
```